### ISSUE TITLE
Bring in hierarchy fixes/bugfixes made in `openforcefield` once infrastructure ready

### ISSUE THREAD

**Issue by davidlmobley** [openforcefield/smirnoff99Frosst#94]
Currently we are waiting to finalize the new versioning/spec infrastructure (by @j-wags ) before making any very significant changes to the FFs here, so this issue is temporarily on hold. But once that infrastructure is in place, we need to bring in the changes listed on `openforcefield` in this issue https://github.com/openforcefield/openforcefield/issues/367 and being brought in via PR https://github.com/openforcefield/openforcefield/pull/368 .

**Comment by j-wags** [comment]
This appears to have been closed by https://github.com/openforcefield/smirnoff99Frosst/pull/99
